cmds/modules/noded/main.go

```diff
@@ -202,6 +205,41 @@ func action(cli *cli.Context) error {
 		return errors.Wrap(err, "failed to create a new perfMon")
 	}
 
+	if exists := iperf.Exists(zinit.Default()); exists {
+		g, err := graphql.NewGraphQl(env.GraphQL)
+		if err != nil {
+			return errors.Wrap(err, "failed to create a new graphql")
+		}
+
+		freeFarmNodes, err := g.ListPublicNodes(0, 1, true, false)
+		if err != nil {
+			return errors.Wrap(err, "failed to list freefarm nodes from graphql")
+		}
+
+		nodes, err := g.ListPublicNodes(12, 0, true, false)
+		if err != nil {
+			return errors.Wrap(err, "failed to list nodes from graphql")
+		}
+
+		nodes = append(nodes, freeFarmNodes...)
+
+		for _, node := range nodes {
+			perfMon.AddTask(&perf.TCPTask{
+				TaskID:    "TCPTask",
+				Schedule:  "* */5 * * * *",
+				ClientIP:  strings.SplitN(node.PublicConfig.Ipv4, "/", 2)[0],
+				Bandwidth: "1M",
```

Comment:
While it's correct, but I really don't like this way to construct types. Instead you should create a `NewTest` function that takes no argument (in this case) because all this stuff should be defined by the test itself, and should be modified by the test plugin (for example iperf test has specific fixed id and schedule)

Also we should have one single test for `iperf` not 2. It's a single test plugin that will do both tcp and udb (sequentially) and store the result of both tests. 

I think a task id can be only `iperf`. I don't know why it's called a Task here when it's not

Reply:
The tests also need to run against MANY nodes per test not a single node. I thought this is clear. so a test can take quite some time before it fiinishes (running multiple tests against multiple nodes for tcp and udp